### ISSUE TITLE
SVG to GIF export broken since version 7.14.5

### ISSUE THREAD

**Issue by denious** [dlemstra/Magick.NET#632]
### Description
I have tried all versions released since 7.14.5 and all result in half-rendering and CPU hogging.

### Steps to Reproduce
Run the code below using [this example SVG](https://1drv.ms/u/s!AtihpW2aJfOWsNBpmvSpz8zK4s5p-Q?e=yKhb02) and notice how the text at the bottom is not rendered correctly in any version above 7.14.5.

7.14.5 rendering result: [view](https://1drv.ms/u/s!AtihpW2aJfOWsNBrJFaXxQIKNJuYfg?e=6ObHfF)
7.15.0+ rendering result: [view](https://1drv.ms/u/s!AtihpW2aJfOWsNBsuwJFqNPSCfsEdA?e=ZAsmjA)

CPU issues are harder to replicate; I published this code to an Azure Web App and everything works fine for the first N conversions, At a certain point the CPU gets blocked and all requests time out. I ran a dump analysis, [results are here for download](https://1drv.ms/u/s!AtihpW2aJfOWsNBqInPI-Qfo4ciQ3g?e=dKJfeT).

```csharp
namespace ImageConverter
{
    public class Converter
    {
        private const int RequiredWidth = 1600;
        private const int MinDpi = 300;

        public static byte[] SvgToGif(Stream svgFileStream)
        {
            using (var svgToGif = new MagickImage())
            {
                // read SVG stream
                svgToGif.Read(svgFileStream, new MagickReadSettings
                {
                    Font = "arial.ttf",
                    Format = MagickFormat.Msvg
                });

                svgToGif.Density = new Density(MinDpi);
                svgToGif.Trim();
                svgToGif.RePage();

                if (svgToGif.Width != RequiredWidth)
                {
                    var resizePercentage = 100 / ((double)svgToGif.Width / RequiredWidth);
                    svgToGif.Resize(new Percentage(resizePercentage));
                }

                // convert new specs to GIF
                svgToGif.Format = MagickFormat.Gif;
                return svgToGif.ToByteArray();
            }
        }
    }
}
```

### System Configuration
- Magick.NET version: > 7.14.5
- Environment (Operating system, version and so on): win-x86 / win-x64

**Comment by denious** [comment]
Here is another [dump analysis for download](https://1drv.ms/u/s!AtihpW2aJfOWsNBtwa87lrOAqE1zWw?e=pvPdoD).
**Magick_Native_Q8_x86!MagickRectangle_Initialize** is last in calls before the system hangs.

```
Thread ID: 18 | Total CPU Time: 13:17:54.999 | Entry Point for Thread: coreclr!Thread::intermediateThreadProc
-- | -- | --
```

```
.NET Call Stack

DomainBoundILStubClass.IL_STUB_PInvoke(IntPtr, ReadWriteStreamDelegate, SeekStreamDelegate, TellStreamDelegate, IntPtr ByRef)+ae
[[InlinedCallFrame]]
ImageMagick.MagickImage+NativeMagickImage.ReadStream(ImageMagick.MagickSettings, ReadWriteStreamDelegate, SeekStreamDelegate, TellStreamDelegate)+58
ImageMagick.MagickImage.Read(System.IO.Stream, ImageMagick.MagickReadSettings, Boolean)+128
SvgConversionUi.Converter.SvgToGif(System.IO.Stream)+a8
```

```
Full Call Stack

Magick_Native_Q8_x86!MagickRectangle_Initialize+14e6fe 
Magick_Native_Q8_x86!MagickRectangle_Initialize+5b763 
Magick_Native_Q8_x86!MagickRectangle_Initialize+1df19e 
0x038db110 
0x118210b8 
0x12158518 
DomainBoundILStubClass.IL_STUB_PInvoke(IntPtr, ReadWriteStreamDelegate, SeekStreamDelegate, TellStreamDelegate, IntPtr ByRef)+ae 
[[InlinedCallFrame]] 
ImageMagick.MagickImage+NativeMagickImage.ReadStream(ImageMagick.MagickSettings, ReadWriteStreamDelegate, SeekStreamDelegate, TellStreamDelegate)+58 
ImageMagick.MagickImage.Read(System.IO.Stream, ImageMagick.MagickReadSettings, Boolean)+128 
SvgConversionUi.Converter.SvgToGif(System.IO.Stream)+a8 
```

**Comment by dlemstra** [comment]
Is it possible to create a console application that reproduces this issue? Maybe with a for loop?

**Comment by dlemstra** [comment]
We can reproduce the problem with the text not being written. Will need to figure out what is causing this and how to fix that issue. It does look like this problem was introduced in version `7.15.5.0` the rest of the versions between `7.14.0.0` and `7.15.5.0` don't seem to have this problem.

**Comment by dlemstra** [comment]
A new release that includes a fix for your issues has been published (17.0.0.1). Can you test if you are still experiencing the CPU issue with that new release?

**Comment by denious** [comment]
Deployed the latest version on Monday, been running good!
